incubator/hnc/third_party/open-policy-agent/gatekeeper/pkg/webhook/certs.go

```diff
@@ -54,23 +56,23 @@ func init() {
 	flag.BoolVar(&restartOnSecretRefresh, "cert-restart-on-secret-refresh", false, "Kills the process when secrets are refreshed so that the pod can be restarted (secrets take up to 60s to be updated by running pods)")
 }
 
-// AddRotator adds the CertRotator and ReconcileVWH to the manager.
-func AddRotator(mgr manager.Manager, cr *CertRotator, vwhName string) error {
+// AddRotator adds the CertRotator and ReconcileWH to the manager.
+func AddRotator(mgr manager.Manager, cr *CertRotator, rwh ReconcileWH) error {
 	cr.client = mgr.GetClient()
 	cr.certsNotMounted = make(chan struct{})
 	if err := mgr.Add(cr); err != nil {
 		return err
 	}
 
-	vwhKey := types.NamespacedName{Name: vwhName}
-	reconciler := &ReconcileVWH{
+	reconciler := &ReconcileWH{
 		client:    mgr.GetClient(),
 		scheme:    mgr.GetScheme(),
 		ctx:       context.Background(),
 		secretKey: cr.SecretKey,
-		vwhKey:    vwhKey,
+		VWHKey:    rwh.VWHKey,
+		CRDNames:  rwh.CRDNames,
```

Comment:
I definitely find it strange that this function accepts one object of type `ReconcileWH` and then creates another completely separate object of the same type to use "for real." I think this is good evidence that `ReconcileWH` should be private and not used outside this package.

Maybe `reconcileWH` could have a pointer to the `CertRotator` so it doesn't have to copy all its fields?